src/App.tsx

```diff
@@ -1,39 +1,59 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerClockNameId = 0;
+
+  componentDidMount(): void {
```

Comment:
It's extra, bc TypeScript already knows about React types
```suggestion
  componentDidMount() {
```

Reply:
That was automaticly added by VS Code